deal_finder/utils/url.py

```diff
@@ -0,0 +1,31 @@
+import re
+
+class URLUtils:
+    @staticmethod
+    def __is_complete_match(regex: re.Pattern[str], url: str) -> bool:
+        if re.fullmatch(regex, url):
+            return True
+        return False
+
+    @staticmethod
+    def is_valid_url(url: str) -> bool:
+        # TODO: Validate the regex string as this was just picked randomly
+        valid_url_regex_string = r"(http|https)://(www.)?[]{2,256}\\.[a-z]{2,6}\\b[-a-zA-Z0-9@:%._\\+~#?&//=]*"
+        valid_url_regex = re.compile(valid_url_regex_string)
+
+        return URLUtils.__is_complete_match(valid_url_regex, url)
+
+    @staticmethod
+    def is_amazon_url(url: str) -> bool:
```

Comment:
can we combine these two methods in one ? is_amazon_url and is_flipkart_url

Reply:
These methods are needed to differentiate b/w url and help us handle the urls accordingly.\

```py
if is_amazon_url():
   get_amazon_meta()
elif is_flipkart_url():
  get_flipkart_meta()
else:
  raise Exception("Unhandled URL")
```

Something on those lines. So the separation is needed.

Reply:
for eg :
if platform == "amazon":
            url_regex_string = r"https://www.amazon.in/[a-zA-Z0-9@:%._\\+~#?&//=,-]*"
elif platform == "flipkart":
            url_regex_string = r"https://www.flipkart.com/[a-zA-Z0-9@:%._\\+~#?&//=,-]*"